New release: Python3 + Cloud + Responsiv + etc

the-maux opened the pull request:
* Migrate Python2.7 to Python3.7 and upgrading all python library
    * Removed libraries:
        * mock==1.0.1
        * requests==1.2.3
        * six>=1.4.1
        * tornado-redis==2.4.2
        * tornado==3.0.1
        * docopt==0.6.1
        * raven==3.5.2
        * Sphinx==1.2.2
        * sphinxcontrib-httpdomain==1.3.0

    * Added libraries:
        * django>=2.2.1
        * django-prometheus>=1.0.15
        * django-cors-headers>=3.0.1
        * channels>=2.2.0
        * channels_redis>=2.4.0
        * asgiref>=3.1.4

* Using Django2.0 over tornado now
    * Updating tornado template tiles-.html to template django, moving them in templates directory
    * Using channels_redis.core.RedisChannelLayer over tornadoredis
    * Using channels.generic.websocket over tornado.websocket.WebSocketHandler
    * Using static files from django behavior over tornado.web.StaticFileHandler
    * Moving tornado url routing behavior to the django behavior
    * removing duplicate inner function to a single file utils.py
    * reworking of the file redis_utils.py to a cache.py
    * Config for ASGI_APPLICATION

* Moving the .tipboard directory to ./tipboard/Config/ to make it compatible with bitnami/python
* Adding dockerfile support
    * Using bitnami image in order to be cloud ready
    * Using non root image
* Adding .gitlab-ci to CI with gitlab
* Minor fixes, improvements, cleanups etc.

the-maux commented:
Update:
* Fixed issue who caused latency with websocket on first load
* Added CI on docker hub here : https://hub.docker.com/r/themaux/tipboard
* Added github package https://github.com/the-maux/tipboard/packages/14519
* Update Readme to show quick usage example (w.i.p)
* Update unit-test (w.i.p)

the-maux commented:
Update:
https://github.com/the-maux/tipboard/commit/20db63eae1452fdff2ae6af4969540204e20a3d8 to answer at issue #21 
features(api): meta & update tile on same request /update
For example, before, your data for the request was:
```json  
  var = {"tile": "big_value",
           "key":  "mybigvalue",
           "data": {"title": "% of something",
                               "description": "% of subtitle",
                               "big-value": "50%",
                               "lower-left-label": "title_value1:",
                               "upper-left-label": "title_value2:",
                               "upper-right-label": "value1",
                               "lower-right-label": "value2"}
         }
```
It's now:
```json  
  var = {"tile": "big_value",
           "key":  "mybigvalue",
           "value": {"big_value_color": "green", "fading_background": "true"},
           "data": {"title": "% of something",
                               "description": "% of subtitle",
                               "big-value": "50%",
                               "lower-left-label": "title_value1:",
                               "upper-left-label": "title_value2:",
                               "upper-right-label": "value1",
                               "lower-right-label": "value2"}
           }
```
But you can still use /push & /tileconfig

jbarbera commented:
I m still having the "" name 'execfile' is not defined "" issue trying to create an empty project, am i missing something? working with python 3.6

the-maux commented:
What command do you use ?
If you use pip, the package tipboard was not update.
You don't need to create an empty project now, by source just type:

> $> python manage runserver 

Python 3.7 is required now

jbarbera commented:
cool , thanks i ll give it a try :D

the-maux commented:
Hey @jbarbera
After 7ac7002 you can now install with pip
```shell
$> virtualenv -p python3 tb-env
$> source tb-env/bin/activate
(tb-env)$> pip install tipboard2.0
(tb-env)$> tipboard runserver 0.0.0.0:8080
```
**It's in WIP** to modify the layout_config.yml, but the package but will be automaticaly update on pypi when fixed, thanks to the [pipeline circle-ci](https://circleci.com/gh/the-maux/tipboard/38?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link)   
Status Badge: [![CircleCI](https://circleci.com/gh/the-maux/tipboard/tree/develop.svg?style=svg)](https://circleci.com/gh/the-maux/tipboard/tree/develop)

fatalcoder commented on CHANGES.rst:
```diff
@@ -1,6 +1,54 @@
 Change Log
 ----------
 
+1.5.0
```
readme is for version 1.5 but `src/__init__.py` `__version__` says 2.0.0

the-maux replied:
Updated with 335d1c1

the-maux reviewed (commented):
first check

the-maux commented:
Closing the PR

the-maux committed:
Codacy(Quality)

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
Merge remote-tracking branch 'origin/wip' into wip

# Conflicts:
#	requirements.txt

the-maux committed:
Quality(Codacy) JS quality

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
Quality(Codacy) JS quality

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
Merge pull request #69 from the-maux/wip

Wip

the-maux committed:
fix(bar_chart) fix gridlines meta +Codacy

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
fix(Codacy) js

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
fix(Falek8) code quality fakeChart

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
Merge pull request #70 from the-maux/wip

Wip

the-maux committed:
Merge pull request #21 from the-maux/develop

develop -> release :D

the-maux committed:
Added tipboard.drawio

the-maux committed:
Update tipboard.drawio

the-maux committed:
Update tipboard.drawio

the-maux committed:
Update tipboard.drawio

the-maux committed:
Update tipboard.drawio

the-maux committed:
Update tipboard.drawio

the-maux committed:
Update tipboard.drawio

the-maux committed:
Update tipboard.drawio

the-maux committed:
Update tipboard.drawio

the-maux committed:
Update tipboard.drawio

the-maux committed:
Update tipboard.drawio

the-maux committed:
Update tipboard.drawio

the-maux committed:
fix(tiles) custom flip timer

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
features(UnitTest) Use Sensors to test all tipboard mecanics

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
add demo.gif

the-maux committed:
Update README.md

the-maux committed:
Add files via upload

pyup-bot committed:
Update pytest from 5.3.4 to 5.3.5

ImgBotApp committed:
[ImgBot] Optimize images

/src/tipboard/tipboard.svg -- 2,049.93kb -> 2,048.77kb (0.06%)

Signed-off-by: ImgBotApp <ImgBotHelp@gmail.com>

the-maux committed:
Update README.md

the-maux committed:
Update README.md

the-maux committed:
fix(Coverage) to 80%, + removing # pragma: no cover

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
fix(Coverage) to 88%: adding Text tiles

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
fix(tiles) custom flip timer

Signed-off-by: max <maxime.renaud@epitech.eu>

pyup-bot committed:
Update redis from 3.3.11 to 3.4.0

the-maux committed:
Merge pull request #73 from the-maux/pyup-update-redis-3.3.11-to-3.4.0

Update redis to 3.4.0

the-maux committed:
Merge pull request #71 from the-maux/pyup-update-pytest-5.3.4-to-5.3.5

Update pytest to 5.3.5

the-maux committed:
Merge pull request #72 from the-maux/imgbot

[ImgBot] Optimize images

the-maux committed:
fix(Unit Test)

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
fix(CircleCI) add redis service

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
fix(CircleCI) fix redis

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
fix(tiles) custom flip timer

Signed-off-by: max <maxime.renaud@epitech.eu>

pyup-bot committed:
Update redis from 3.4.0 to 3.4.1

the-maux committed:
Update README.md

pyup-bot committed:
Update django from 3.0.2 to 3.0.3

the-maux committed:
fix(JS) refacto

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
broke-me(iframe) fuck off

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
broke-me(iframe) fuck off

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
Add files via upload

ImgBotApp committed:
[ImgBot] Optimize images

/src/tipboard/tipboard.svg -- 2,049.93kb -> 2,048.77kb (0.06%)

Signed-off-by: ImgBotApp <ImgBotHelp@gmail.com>

the-maux committed:
broke-me(iframe) p2

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
broke-me(iframe) p2

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
broke-me(iframe) p2

Signed-off-by: max <maxime.renaud@epitech.eu>

dependabot[bot] committed:
Bump django from 3.0.2 to 3.0.3

Bumps [django](https://github.com/django/django) from 3.0.2 to 3.0.3.
- [Release notes](https://github.com/django/django/releases)
- [Commits](https://github.com/django/django/compare/3.0.2...3.0.3)

Signed-off-by: dependabot[bot] <support@github.com>

the-maux committed:
wip(b&w) p1

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
wip(b&w) .css cleaning

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
wip(b&w) just a message for Allegro guy <3

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
wip(b&w) p3: color_mode in .yaml

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
Merge pull request #75 from the-maux/pyup-update-django-3.0.2-to-3.0.3

Update django to 3.0.3

the-maux committed:
Merge pull request #77 from the-maux/dependabot/pip/django-3.0.3

Bump django from 3.0.2 to 3.0.3

the-maux committed:
Merge pull request #74 from the-maux/pyup-update-redis-3.4.0-to-3.4.1

Update redis to 3.4.1

the-maux committed:
Update README.md

the-maux committed:
Update README.md

the-maux committed:
Update README.md

the-maux committed:
Update README.md

the-maux committed:
Delete TODO.md

the-maux committed:
broke-me(iframe) p2

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
fix quality

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
fix quality

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
quality Codacy fix

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
Merge pull request #78 from the-maux/black_and_white_finaly

feat(b&w) -> wip

the-maux committed:
Merge branch 'wip' into develop

the-maux committed:
Merge remote-tracking branch 'origin/develop' into develop

# Conflicts:
#	TODO.md

the-maux committed:
fix(Github Actions) add redis service

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
fix(G.A) add redis with zhulik technique

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
oups

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
broke-me(iframe) p2

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
AHHH

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
fix(G.A) full test

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
...

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
fix C.I

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
fix(travis-ci) test add redis

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
fix(circle-CI) syntax

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
circleci test

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
fix circleci

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
broke-me(iframe) p2

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
broke-me(iframe) p2

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
wip(gauge) add radial gauge

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
wip(weight) by parser.py p2

Signed-off-by: max <maxime.renaud@epitech.eu>

pyup-bot committed:
Update requests from 2.22.0 to 2.23.0

pyup-bot committed:
Update channels_redis from 2.4.1 to 2.4.2

the-maux committed:
gauge wip

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
fix(ImgBot) svg to png

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
Merge pull request #76 from the-maux/imgbot

[ImgBot] -> develop : Optimize .svg

the-maux committed:
Merge remote-tracking branch 'origin/develop' into develop

the-maux committed:
Merge pull request #81 from the-maux/pyup-update-requests-2.22.0-to-2.23.0

Update requests to 2.23.0

the-maux committed:
Merge pull request #82 from the-maux/pyup-update-channels_redis-2.4.1-to-2.4.2

Update channels_redis to 2.4.2

the-maux committed:
fix(demo) when multiple start issue #86

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
quality(linearradial) var -> let

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
wip

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
fix(parser) missing template_name

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
quality(flake8 + js)

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
quality(flake8 + js)

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
quality(flake8 + js)

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
Merge pull request #84 from the-maux/wip

Wip

the-maux committed:
quality(flake8 + js)

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
Merge pull request #87 from the-maux/wip

quality(flake8 + js)

the-maux committed:
quality(flake8 + js)

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
wip(deconnection) adding alert when server disconnect

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
wip(w-row:magic) + refacto parser.py

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
wip(parser) refacto

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
wip(deconnection) refacto api.py & parser.py

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
quality(js) code quality

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
wip(linear_gauge) adding horizontal linear gauge

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
quality(cache) quality & complexity

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
Merge pull request #89 from the-maux/wip

Wip

the-maux committed:
Add files via upload

ImgBotApp committed:
[ImgBot] Optimize images

*Total -- 695.42kb -> 582.33kb (16.26%)

/src/tipboard/docs/ressources/gcp_logo.png -- 10.18kb -> 4.00kb (60.66%)
/src/tipboard/docs/ressources/githubactions_actions.png -- 9.86kb -> 4.62kb (53.21%)
/src/tipboard/docs/ressources/circle_ci_logo.png -- 19.53kb -> 10.78kb (44.81%)
/src/tipboard/docs/ressources/openshift_logo.png -- 8.93kb -> 6.65kb (25.59%)
/src/tipboard/docs/ressources/codacy_logo.png -- 7.43kb -> 5.59kb (24.72%)
/src/tipboard/docs/ressources/docker_logo.png -- 87.93kb -> 66.61kb (24.25%)
/src/tipboard/docs/ressources/openshift-logo.jpg -- 67.74kb -> 56.07kb (17.24%)
/src/tipboard/docs/ressources/azure_logo.png -- 49.56kb -> 42.22kb (14.81%)
/src/tipboard/docs/ressources/codacy_logo.jpg -- 76.68kb -> 65.73kb (14.28%)
/src/tipboard/docs/ressources/tile_ex.png -- 95.18kb -> 82.36kb (13.46%)
/src/tipboard/tipboard.png -- 126.37kb -> 110.93kb (12.22%)
/src/tipboard/docs/ressources/django_logo.png -- 5.88kb -> 5.20kb (11.48%)
/src/tipboard/docs/ressources/codebeat_logo.png -- 3.89kb -> 3.52kb (9.57%)
/src/tipboard/docs/ressources/better-code-hub.png -- 42.28kb -> 38.40kb (9.17%)
/src/tipboard/docs/ressources/codeinspector_logo.png -- 57.19kb -> 53.34kb (6.74%)
/src/tipboard/docs/ressources/code_climate.png -- 5.36kb -> 5.17kb (3.62%)
/src/tipboard/static/img/logo-tipboard.svg -- 4.21kb -> 4.09kb (2.94%)
/src/tipboard/docs/ressources/github-package-registry-min.png -- 5.38kb -> 5.30kb (1.4%)
/src/tipboard/docs/ressources/scrutinizer_logo.png -- 3.25kb -> 3.23kb (0.72%)
/src/tipboard/docs/ressources/librarie_is.jpg -- 8.58kb -> 8.52kb (0.71%)

Signed-off-by: ImgBotApp <ImgBotHelp@gmail.com>

the-maux committed:
Add files via upload

the-maux committed:
Add files via upload

the-maux committed:
Add files via upload

the-maux committed:
Add files via upload

the-maux committed:
wip(deconnection) adding alert when server disconnect

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
Merge pull request #90 from the-maux/imgbot

[ImgBot] Optimize images

the-maux committed:
Add files via upload

ImgBotApp committed:
[ImgBot] Optimize images

*Total -- 102.36kb -> 76.38kb (25.38%)

/src/tipboard/docs/ressources/white_mode.png -- 46.43kb -> 32.05kb (30.97%)
/src/tipboard/docs/ressources/tipboard_API_PUSH.png -- 51.72kb -> 40.25kb (22.19%)
/src/tipboard/docs/ressources/logo-tipboard.svg -- 4.21kb -> 4.09kb (2.94%)

Signed-off-by: ImgBotApp <ImgBotHelp@gmail.com>

the-maux committed:
feat(stream tile) adding stream tile

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
Merge pull request #92 from the-maux/imgbot

[ImgBot] Optimize images

the-maux committed:
quality

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
feat(css) improve CSS

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
clean(git) remove unnecessary file

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
clean(git) remove unnecessary file

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
clean(git) remove unnecessary file

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
clean(css) layout.css

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
quality(flake8+js) multiple from Codacy & Deepscan

Signed-off-by: max <maxime.renaud@epitech.eu>

pyup-bot committed:
Update django from 3.0.3 to 3.0.4

the-maux committed:
Test(fix) parser & Flipboard unit-test

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
Merge pull request #94 from the-maux/pyup-update-django-3.0.3-to-3.0.4

Update django to 3.0.4

the-maux committed:
Test(fix) adding basic layout_config.yaml

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
Test(fix) adding basic layout_config.yaml

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
quality(fix) codefactor

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
feat(custom tile) adding template

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
Add files via upload

the-maux committed:
Delete circular_gauge

ImgBotApp committed:
[ImgBot] Optimize images

*Total -- 47.81kb -> 36.23kb (24.23%)

/src/tipboard/docs/ressources/radial_gauge.png -- 23.90kb -> 17.54kb (26.6%)
/src/tipboard/docs/ressources/vertical_linear_gauge.png -- 12.21kb -> 9.42kb (22.85%)
/src/tipboard/docs/ressources/linear_gauge.png -- 11.70kb -> 9.26kb (20.82%)

Signed-off-by: ImgBotApp <ImgBotHelp@gmail.com>

the-maux committed:
Add files via upload

the-maux committed:
wip(custom_tile) init custom tile

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
wip(Custom tile) first commit

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
wip(Custom tile) first commit

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
Merge pull request #96 from the-maux/imgbot

[ImgBot] Optimize images

the-maux committed:
quality(Custom tile) text_value.js

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
quality(Custom tile) secu django 3.0.4

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
quality(Custom tile) quality code

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
quality(Custom tile) quality code

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
quality(Custom tile) quality code

Signed-off-by: max <maxime.renaud@epitech.eu>

pyup-bot committed:
Update asgiref from 3.2.3 to 3.2.4

the-maux committed:
Merge pull request #98 from the-maux/pyup-update-asgiref-3.2.3-to-3.2.4

Update asgiref to 3.2.4

the-maux committed:
quality(Custom tile) quality code

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
quality(Custom tile) fix flake8

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
quality(Custom tile) fix flake8

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
Merge pull request #95 from the-maux/feat/custom_tiles

feat(custom tile) adding custom tile

the-maux committed:
remove gitlab-ci bc i'm lazy to add redis support in C.I test -_-

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
fix(LAYOUT_CONFIG) first fix

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
merge security with branch dev

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
Merge branch 'fix/properties-LAYOUT_CONFIG' into wip

the-maux committed:
fix(merge) LAYOUT_CONFIG

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
fix(FLIPBOARD_INTERVAL) fix FLIPBOARD_INTERVAL

Signed-off-by: max <maxime.renaud@epitech.eu>

pyup-bot committed:
Update asgiref from 3.2.4 to 3.2.5

the-maux committed:
fix(C.I) stop gitlab-ci & improve gitignore for comfort

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
fix(iframe) add sensors exemple

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
Merge pull request #102 from the-maux/pyup-update-asgiref-3.2.4-to-3.2.5

Update asgiref to 3.2.5

the-maux committed:
quality(text_value) quick codebeat

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
C.I(CodeClimate) Improve config

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
quality(CodeBEAT) tileStream ABC

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
quality(CodeClimate) clear

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
quality(CodeClimate) clear

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
add setup.cfg

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
Merge pull request #99 from the-maux/wip

Wip -> develop

the-maux committed:
quality Codebeat

Signed-off-by: max <maxime.renaud@epitech.eu>

ImgBotApp committed:
[ImgBot] Optimize images

/src/tipboard/static/img/disconnect.png -- 35.89kb -> 26.27kb (26.8%)

Signed-off-by: ImgBotApp <ImgBotHelp@gmail.com>

the-maux committed:
Merge pull request #103 from the-maux/imgbot

[ImgBot] Optimize images

pyup-bot committed:
Update pytest from 5.3.5 to 5.4.0

the-maux committed:
Merge pull request #104 from the-maux/pyup-update-pytest-5.3.5-to-5.4.0

Update pytest to 5.4.0

pyup-bot committed:
Update pytest from 5.4.0 to 5.4.1

the-maux committed:
Merge pull request #105 from the-maux/pyup-update-pytest-5.4.0-to-5.4.1

Update pytest to 5.4.1

the-maux committed:
fix(ChartJS) update chart meta not working on ws update

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
flake8(fix) fix issue

Signed-off-by: max <maxime.renaud@epitech.eu>

pyup-bot committed:
Update pyyaml from 5.3 to 5.3.1

the-maux committed:
Merge pull request #107 from the-maux/pyup-update-pyyaml-5.3-to-5.3.1

Update pyyaml to 5.3.1

the-maux committed:
Add files via upload

ImgBotApp committed:
[ImgBot] Optimize images

*Total -- 32.67kb -> 22.43kb (31.35%)

/src/tipboard/docs/ressources/percentage_ex.png -- 16.74kb -> 11.45kb (31.63%)
/src/tipboard/docs/ressources/justvalue_ex.png -- 15.93kb -> 10.98kb (31.07%)

Signed-off-by: ImgBotApp <ImgBotHelp@gmail.com>

the-maux committed:
Add files via upload

KbYll committed:
Update sensors_main.py for schedule time

Scheduler work in 6 seconds all times.

Now, we can schedule job with the needed time

pyup-bot committed:
Update asgiref from 3.2.5 to 3.2.7

the-maux committed:
Merge pull request #111 from KbYll/patch-1

Update sensors_main.py for schedule time

the-maux committed:
Merge pull request #109 from the-maux/imgbot

[ImgBot] Optimize images

the-maux committed:
Merge pull request #113 from the-maux/pyup-update-asgiref-3.2.5-to-3.2.7

Update asgiref to 3.2.7

snyk-bot committed:
fix: requirements.txt to reduce vulnerabilities


The following vulnerabilities are fixed by pinning transitive dependencies:
- https://snyk.io/vuln/SNYK-PYTHON-PYYAML-559098

the-maux committed:
Merge pull request #114 from the-maux/snyk-fix-098b8b2373751faf4a480136d4bb996d

[Snyk] Security upgrade pyyaml from 5.3 to 5.3.1

the-maux committed:
Add files via upload

ImgBotApp committed:
[ImgBot] Optimize images

*Total -- 113.25kb -> 93.07kb (17.81%)

/src/tipboard/docs/ressources/text_ex.png -- 16.69kb -> 9.73kb (41.74%)
/src/tipboard/docs/ressources/listing_ex.png -- 12.18kb -> 8.14kb (33.21%)
/src/tipboard/docs/ressources/customhtml.png -- 84.37kb -> 75.21kb (10.86%)

Signed-off-by: ImgBotApp <ImgBotHelp@gmail.com>

the-maux committed:
Merge pull request #116 from the-maux/imgbot

[ImgBot] Optimize images

the-maux committed:
fix(custom_tile) add real exemple

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
test(sensors_main) test if all sensors test are present

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
Merge branch 'develop' into wip

the-maux committed:
fix(travis) remove python 3.6 & 3.7 support, only 3.8

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
Merge remote-tracking branch 'origin/wip' into wip

the-maux committed:
Merge pull request #106 from the-maux/wip

wip -> develop

the-maux committed:
Merge pull request #117 from the-maux/develop

Merge pull request #106 from the-maux/wip

the-maux committed:
Merge branch 'master' into develop

pyup-bot committed:
Update pytest-django from 3.8.0 to 3.9.0

pyup-bot committed:
Update django from 3.0.4 to 3.0.5

the-maux committed:
Merge pull request #118 from the-maux/pyup-update-pytest-django-3.8.0-to-3.9.0

Update pytest-django to 3.9.0

the-maux committed:
Merge pull request #119 from the-maux/pyup-update-django-3.0.4-to-3.0.5

Update django to 3.0.5

the-maux committed:
quality(Redis) use a real singleton, to remove global variable

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
quality(LAYOUT_CONFIG)

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
clean

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
quality(depencies) remove BDD support & CORS support

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
quality(cache) remove broken deserialiazation

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
fix(test)

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
improve coverage

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
fix(chartJS) fix broken tile update

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
fix(api) bug when update meta

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
features(api) add infos in API

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
Merge pull request #120 from the-maux/wip

Wip

the-maux committed:
wip

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
fix little refacto

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
fix flake8

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
quality(logger) improve print

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
features(white-noise) enabling production mode

Signed-off-by: max <maxime.renaud@epitech.eu>

the-maux committed:
Merge pull request #121 from the-maux/wip

wip

the-maux commented:
**679 commit later**, i'm back and almost ready to publish Tipboard 3.0,
I will start a new depot, named Chartboard under M.I.T License (except if we merge this PR :smiley:).
I will soon archived my working depot, to start a full new project with a new name (like piwik for matomo :laughing:)
Asap, I will put a full list of changes in the Change.rst and here, also[ the migration story](https://github.com/the-maux/tipboard/wiki/Tipboard-migration) in the wiki
Don't hesitate to check the new [wiki](https://github.com/the-maux/tipboard/wiki)
I hope you will enjoy it, there are a lot of new features :)

pyup-bot committed:
Update redis from 3.4.1 to 3.5.0

pyup-bot committed:
Update pytest-asyncio from 0.10.0 to 0.12.0

pyup-bot committed:
Update django from 3.0.5 to 3.0.6

the-maux committed:
Merge pull request #125 from the-maux/pyup-update-redis-3.4.1-to-3.5.0

Update redis to 3.5.0

the-maux committed:
Merge pull request #126 from the-maux/pyup-update-pytest-asyncio-0.10.0-to-0.12.0

Update pytest-asyncio to 0.12.0

the-maux committed:
Merge pull request #127 from the-maux/pyup-update-django-3.0.5-to-3.0.6

Update django to 3.0.6

pyup-bot committed:
Update pytest from 5.4.1 to 5.4.2

pyup-bot committed:
Update redis from 3.5.0 to 3.5.1

the-maux committed:
Merge pull request #128 from the-maux/pyup-update-pytest-5.4.1-to-5.4.2

Update pytest to 5.4.2

the-maux committed:
Merge pull request #129 from the-maux/pyup-update-redis-3.5.0-to-3.5.1

Update redis to 3.5.1

the-maux committed:
Update README.md

pyup-bot committed:
Update redis from 3.5.1 to 3.5.2

pyup-bot committed:
Update django-cors-headers from 3.2.1 to 3.3.0

pyup-bot committed:
Update whitenoise from 5.0.1 to 5.1.0

the-maux committed:
Merge pull request #130 from the-maux/pyup-update-redis-3.5.1-to-3.5.2

Update redis to 3.5.2

the-maux committed:
Merge pull request #132 from the-maux/pyup-update-django-cors-headers-3.2.1-to-3.3.0

Update django-cors-headers to 3.3.0

the-maux committed:
Merge pull request #133 from the-maux/pyup-update-whitenoise-5.0.1-to-5.1.0

Update whitenoise to 5.1.0

vi4m commented:
Thanks for keeping working on this fork, looks really promising.

the-maux commented:
@vi4m thanks 😄 
Don't hesitate to check the new fork https://github.com/the-maux/Chartboard 
Bc Covid, i couldn't work like i wanted on it since, but it's almost production ready 
I think i need help on documentation, **i'm so fucking lazy about it** !
But i will do it, Chartboard v1.0 wil be, and maybe one day we will full merge ❤️

the-maux committed:
Update README.md

the-maux committed:
Update README.md

vi4m commented:
How's the work going? Is it ready?

the-maux commented:
I didn't saw your message sorry. 
But yes, Check it out :https://github.com/the-maux/Chartboard
